app/src/main/java/com/example/drawingactivity/DrawActivity.kt

```diff
@@ -89,88 +90,117 @@ class DrawActivity : AppCompatActivity() {
                     })
-                if (it.isDialogOpen && !isEditDialogOpen) {
-                    isEditDialogOpen = true
-                    dialogUtils.showDialog(
-                        context = this,
-                        layerList = it.layerList,
-                        onEditText = { layerName -> viewModel.setCurrentText(layerName) },
-                        previousInput = it.currentNewLayerName,
-                        onSubmit = { newLayerName ->
-                            if (newLayerName != null) {
-                                viewModel.editLayer(newLayerName, it.editLayerIndex)
-                                bottomSheetLambda?.invoke(
-                                    it.editLayerIndex, newLayerName,
-                                    DialogOperation.EDIT
-                                )
-                            } else viewModel.dialogCompleted()
-                        },
-                    )
-                } else if (!it.isDialogOpen) {
-                    isEditDialogOpen = false
-                }
-            } else if (!it.isLayerSheetOpen) {
-                isLayerListViewSheetOpen = false
-            }
+            } else isLayerListViewSheetOpen = false
+            if (it.isDeleteDialogOpen && !isDeleteDialogOpen) {
+                isDeleteDialogOpen = true
+                dialogUtils.confirmationDialog(
+                    context = this@DrawActivity,
+                    title = this@DrawActivity.getString(R.string.deleteLayerDialogTitle),
+                    message = this@DrawActivity.getString(
+                        R.string.deleteLayerDialogBody, it.layerList[it.listIndex]
+                    ),
+                    onConfirm = {
+                        viewModel.removeFromList(it.listIndex)
+                        onLayerListUpdate?.invoke(
+                            viewModel.getCurrentIndex(), null, DialogOperation.DELETE
+                        )
+                        viewModel.dialogCompleted()
+                    },
+                    onCancel = {
+                        viewModel.dialogCompleted()
+                    })
+            } else isDeleteDialogOpen = false
+            if (it.isEditDialogOpen && !isEditDialogOpen) {
+                isEditDialogOpen = true
+                dialogUtils.showDialog(
+                    context = this,
+                    layerList = it.layerList,
+                    onEditText = { layerName -> viewModel.setCurrentText(layerName) },
+                    previousInput = it.currentNewLayerName,
+                    onSubmit = { newName ->
+                        if (newName != null) {
+                            viewModel.editListAtIndex(newName, it.listIndex)
+                            onLayerListUpdate?.invoke(
+                                it.listIndex, newName, DialogOperation.EDIT
+                            )
+                        } else viewModel.dialogCompleted()
```

Comment:
This should just be calling something in the viewmodel that updates its state and marks the dialog completed by itself without multiple calls, like my other comment

Reply:
removed this